data_dictionary/data_dictionary.cc

```diff
@@ -243,6 +261,50 @@ std::vector<view_ptr> keyspace_metadata::views() const {
             | boost::adaptors::transformed([] (auto&& s) { return view_ptr(s); }));
 }
 
+storage_options::value_type storage_options::from_map(std::string_view type, std::map<sstring, sstring> values) {
+    if (type == "LOCAL") {
+        return local{};
+    }
+    if (type == "S3") {
+        s3 options;
+        for (const auto& [key, value]: values) {
+            if (key == "bucket") {
+                options.bucket = value;
+            } else if (key == "endpoint") {
+                options.endpoint = value;
+            } else {
+                throw std::runtime_error(format("Unknown storage option for S3: {}", key));
+            }
+        }
+        return options;
+    }
+    throw std::runtime_error(format("Unknown storage type: {}", type));
+}
+
+std::string_view storage_options::type_string() const {
+    return std::visit(overloaded_functor {
+        [] (const storage_options::local&) {
+            return "LOCAL";
+        },
+        [] (const storage_options::s3&) {
+            return "S3";
+        }
+    }, value);
+}
```

Comment:
I think this implementation is fine, but just want to comment that in the good (?) all days, we used to do this kind of things with inheritance, not std::variant. It allowed a future developer to add a new storage type (e.g., EBS, or whatever) by mostly adding new code in a new file, instead of having to modify a dozen places in an existing file.
But for now, I think what you did is good enough, no need to change.